src/lib/datatypes/include/sol-vector.h

```diff
@@ -198,6 +198,20 @@ sol_vector_del_last(struct sol_vector *v)
 }
 
 /**
+ * @brief Remove an range of element from the vector.
+ *
+ * Removes the range starting at index @a start from the vector and
+ * goes until @a start + @a len.
+ *
+ * @param v Vector pointer
+ * @param start Index of the first element to remove
+ * @param len the number of elements to remover
+ *
+ * @return @c 0 on success, error code (always negative) otherwise
+ */
+int sol_vector_del_range(struct sol_vector *v, uint16_t start, uint16_t len);
```

Comment:
I guess we should add the same function for sol_ptr_vector. Maybe a static inline function that calsl `sol_vector_del_range()`